src/main/java/userReg/UserRegController.java

```diff
@@ -0,0 +1,34 @@
+package userReg;
+
+import data_access_storage.UserRepo;
+import entities.UofTStudentEmailVerifier;
+import entities.User;
+import entities.UserAccount;
+
+import java.io.File;
+import java.io.IOException;
+
+public class UserRegController {
+
+
+    /**
+     *
+     */
+    public String  userProfileCreate(UserProfile up){
+
+        String eMail=up.getEmail();
+
+        if (!up.checkEmail()){
+            return "Err01";
+        }
+
+        if (up.existEmail()){
+            return "Err02";
+        }
+
+        up.save();
+        return "OK";
+
+
+    }
+}

```

Comment:
This class violates dependence rules required by clean architecture. The controller should not skip the interactor and uses classes from entities directly. The job of a controller is  to bundle input from user-at-keyword and sends it to the appropriate use  case